### ISSUE TITLE
Not possible to disable plugin per application variant with Kotlin Gradle DSL

### ISSUE THREAD

**Issue by pbajurko** [bugsnag/bugsnag-android-gradle-plugin#203]
### Expected behavior
From [docs](https://docs.bugsnag.com/build-integrations/gradle/#additional-configuration)

> Disabling Bugsnag
> 
> To completely disable the Bugsnag plugin, use the extension property enableBugsnag, for example:

```
android {
    buildTypes {
        debug {
            ext.enableBugsnag = false
        }
    }
}
```

### Observed behavior
There is no way to configure BuildType or Flavor since they are not [ExtensionAware](https://docs.gradle.org/current/dsl/org.gradle.api.plugins.ExtensionAware.html )

### Steps to reproduce
Setup basic android app with build.gradle.kts

### Version
4.7.4

### Additional information
Bugsnag plugin should expose configuration option e.g. in BugsnagPluginExtension to configure variant filter.

**Comment by pbajurko** [comment]
Actually I stand corrected looks like those AGP types are [ExtensionAware](https://github.com/gradle/kotlin-dsl-samples/issues/1254#issuecomment-486259572)
so solution would be

```
    buildTypes {
        getByName("debug") {
            require(this is ExtensionAware)
            extra["enableBugsnag"] = false
        }
    }
```
Maybe something worth to add to docs?

Still I think cleaner API would be:
```
configure<BugsnagPluginExtension> {
    variantFilter {
        name.contains("release")
    }
}
```

**Comment by johnkiely1** [comment]
Hi @pbajurko, Kotlin Gradle DSL is not something we officially support. We are looking in so this may be something we officially support in the future, so thanks for the suggestion and will pass it on to our products team.

**Comment by ZacSweers** [comment]
I think the `extra["enableBugsnag"]` appraoch is the right way to do it here for now. Another option would be for the plugin's extension to offer an API for filtering variant names

**Comment by ZacSweers** [comment]
This is resolved by #255
